### PR TITLE
Rest cache - property cache

### PR DESCRIPTION
**Pull request by xyuanlu**
### Issues

- [X] My PR addresses the following Helix issues and references them in the PR description:

#2659

### Description

- [X] Here are some details about my PR, including screenshots of any UI changes:

Structure for rest data provider. It supports concurrent read/update.

### Tests

- [X] The following tests are written for this issue:

Test will be added in following prs.

- The following is the result of the "mvn test" command on the appropriate module:

(If CI test fails due to known issue, please specify the issue and test PR locally. Then copy & paste the result of "mvn test" to here.)

### Changes that Break Backward Compatibility (Optional)

- My PR contains changes that break backward compatibility or previous assumptions for certain methods or API. They include:

(Consider including all behavior changes for public methods or API. Also include these changes in merge description so that other developers are aware of these changes. This allows them to make relevant code changes in feature branches accounting for the new method/API behavior.)

### Documentation (Optional)

- In case of new functionality, my PR adds documentation in the following wiki page:

(Link the GitHub wiki you added)

### Commits

- My commits all reference appropriate Apache Helix GitHub issues in their subject lines. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Code Quality

- My diff has been formatted using helix-style.xml 
(helix-style-intellij.xml if IntelliJ IDE is used)

### CODE REVIEW

File: helix-rest/src/main/java/org/apache/helix/rest/common/dataprovider/PerClusterDataProvider.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    _instanceConfigCache = null;
+    _clusterConfigCache = null;
+    _resourceConfigCache = null;
+    _liveInstanceCache = null;
+    _idealStateCache = null;
+    _stateModelDefinitionCache = null;
+    _currentStateCache = null;
+  }
+  // TODO: consolidate EV from CSs
+  public Map<String, ExternalView> consolidateExternalViews() {
```
Comment (MarkGaox) [inline helix-rest/src/main/java/org/apache/helix/rest/common/dataprovider/PerClusterDataProvider.java:79]: Just to understand, if the stoppable check wants to get the external view from the cache, it should call `consolidateExternalViews `which will generate EV by the data from `RestCurrentStateCache`? Is that the plan?

Comment (xyuanlu) [reply helix-rest/src/main/java/org/apache/helix/rest/common/dataprovider/PerClusterDataProvider.java:79]: It should just call getEV. consolidateExternalViews will be called internal in this data provider

File: helix-core/src/test/java/org/apache/helix/integration/rebalancer/TestInstanceOperation.java [hunk 2]
```diff
@@ -338,21 +341,38 @@ public void testMarkEvacuationAfterEMM() throws Exception {
...
+                v -> v.equals("MASTER") || v.equals("LEADER") || v.equals("SLAVE") || v.equals("FOLLOWER") || v.equals(
+                    "STANDBY"))
+            .forEach(v -> activeReplicaCount.getAndIncrement());
+        Assert.assertTrue(activeReplicaCount.get() >= REPLICA-1);
+        Assert.assertFalse(ev.getStateMap(partition).containsKey(evacuateInstanceName) && ev.getStateMap(partition)
+            .get(evacuateInstanceName)
+            .equals("MASTER") && ev.getStateMap(partition)
+            .get(evacuateInstanceName)
+            .equals("LEADER"));
+
```
Comment (MarkGaox) [inline helix-core/src/test/java/org/apache/helix/integration/rebalancer/TestInstanceOperation.java:371]: nit: empty line

**Comment by junkaixue** [comment]
Is this still a valid change?

**Pull request closed without merging**